### PR TITLE
Update JitsiStreamPresenterEffect.js

### PR DESCRIPTION
**Pull request by miticc06**
stop woker _videoFrameTimerWorker

### CODE REVIEW

**Comment by saghul** [comment]
Hum, on a quick glance it looks like this would make the start / stop / start sequence of actions impossible, since the worker is only created once, right?

**Comment by miticc06** [comment]
> Hi, thanks for your contribution!
> If you haven't already done so, could you please make sure you sign our CLA (https://jitsi.org/icla for individuals and https://jitsi.org/ccla for corporations)? We would unfortunately be unable to merge your patch unless we have that piece :(.

When you share the screen, turn the camera on and off repeatedly. You will see that the woker has been created but not canceled.

Screenshot from 2020-07-21 15-35-15

**Comment by miticc06** [comment]
> Hum, on a quick glance it looks like this would make the start / stop / start sequence of actions impossible, since the worker is only created once, right?

When you share the screen, turn the camera on and off repeatedly. You will see that the woker has been created but not canceled.

Screenshot from 2020-07-21 15-35-15

**Comment by saghul** [comment]
@jallamsetty1 Can you PTAL? Also, come to think of it, are we creating more and more hidden elements in the DOM to make the composition?

**Comment by jallamsetty1** [comment]
> @jallamsetty1 Can you PTAL? Also, come to think of it, are we creating more and more hidden elements in the DOM to make the composition?

No we are not creating more and more hidden elements in the DOM as a result of start/stop. They all get created when the effect is created.

**Comment by jallamsetty1** [comment]
Jenkins please test this please.

File: react/features/stream-effects/presenter/JitsiStreamPresenterEffect.js [hunk 1]
```diff
@@ -153,6 +153,7 @@ export default class JitsiStreamPresenterEffect {
         this._videoFrameTimerWorker.postMessage({
             id: CLEAR_INTERVAL
         });
+        this._videoFrameTimerWorker.terminate();
```
Comment (jallamsetty1) [inline react/features/stream-effects/presenter/JitsiStreamPresenterEffect.js:156]: The worker is created only once when the effect is constructed. Worker creation needs to be moved to startEffect.

Comment (miticc06) [reply react/features/stream-effects/presenter/JitsiStreamPresenterEffect.js:156]: > The worker is created only once when the effect is constructed. Worker creation needs to be moved to startEffect.


I have updated my code.

Comment (jallamsetty1) [reply react/features/stream-effects/presenter/JitsiStreamPresenterEffect.js:156]: Thank you, can you please sign the CLA so that we can accept the PR.

**Comment by jallamsetty1** [comment]
> @jallamsetty1 Can you PTAL? Also, come to think of it, are we creating more and more hidden elements in the DOM to make the composition?

@saghul you are right, a new effect is getting created every time we mute/unmute the presenter. I will fix that in conference.js

**Comment by jallamsetty1** [comment]
Jenkins please test this please.

**Comment by miticc06** [comment]
> Hi, thanks for your contribution!
> If you haven't already done so, could you please make sure you sign our CLA (https://jitsi.org/icla for individuals and https://jitsi.org/ccla for corporations)? We would unfortunately be unable to merge your patch unless we have that piece :(.

I signed ILCA. Please check again.

**Comment by jallamsetty1** [comment]
> > Hi, thanks for your contribution!
> > If you haven't already done so, could you please make sure you sign our CLA (https://jitsi.org/icla for individuals and https://jitsi.org/ccla for corporations)? We would unfortunately be unable to merge your patch unless we have that piece :(.
> 
> I signed ILCA. Please check again.

Found it, thanks !

**Pull request merged**